lib/parse.js

```diff
@@ -36,13 +40,20 @@ function parseNonShell(parsed) {
     if (needsShell) {
         // Escape command & arguments
         applyQuotes = (parsed.command !== 'echo');  // Do not quote arguments for the special "echo" command
-        parsed.command = escapeCommand(parsed.command);
-        parsed.args = parsed.args.map(function (arg) {
+        command = escapeCommand(parsed.command);
+        args = parsed.args.map(function (arg) {
             return escapeArgument(arg, applyQuotes);
         });
 
         // Make use of cmd.exe
-        parsed.args = ['/d', '/s', '/c', '"' + parsed.command + (parsed.args.length ? ' ' + parsed.args.join(' ') : '') + '"'];
+        // Pass command as environment variable
+        parsed.options.env = assign({}, parsed.options.env || process.env);
+        do {
```

Comment:
My implementation is technically safer, although I agree the possibility of a collision is remote.  If you're dedicated to saving 2 lines of code then I'll change it, but at this point the code's already written, so I think it should stay the way it is.